internal/reachability/client_test.go

```diff
@@ -0,0 +1,258 @@
+package reachability_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/snyk/cli-extension-os-flows/internal/reachability"
+)
+
+func Test_StartReachabilityAnalysis(t *testing.T) {
+	orgID := uuid.MustParse("b90ce9f4-f35e-4bee-85a2-0aca4feb5593")
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "2024-10-15", r.URL.Query().Get("version"))
+		assert.Equal(t, fmt.Sprintf("/hidden/orgs/%s/reachability", orgID), r.URL.Path)
+		assert.Equal(t, http.MethodPost, r.Method)
+
+		w.WriteHeader(http.StatusCreated)
+		w.Write([]byte(`{
+			"data": {
+				"id": "20fca088-0ac3-48bd-85b1-cdfccd29cea3"
+			}
+		}`))
+	}))
+	defer srv.Close()
+	rc := reachability.NewClient(srv.Client(), reachability.Config{
+		BaseURL: srv.URL,
+	})
+
+	reachID, err := rc.StartReachabilityAnalysis(t.Context(), orgID, "bundle-hash")
+	require.NoError(t, err)
+
+	assert.Equal(t, "20fca088-0ac3-48bd-85b1-cdfccd29cea3", reachID.String())
+}
+
+func Test_StartReachabilityAnalysis_ServerError(t *testing.T) {
+	orgID := uuid.MustParse("b90ce9f4-f35e-4bee-85a2-0aca4feb5593")
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "2024-10-15", r.URL.Query().Get("version"))
+		assert.Equal(t, fmt.Sprintf("/hidden/orgs/%s/reachability", orgID), r.URL.Path)
+		assert.Equal(t, http.MethodPost, r.Method)
+
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(`{"errors": [{"title": "Internal server error"}]}`))
+	}))
```

Comment:
I think it would be nice to refactor this in a helper function which accepts
- status code, response body
- expectePath, expected Method, expected version